pkg/cluster/internal/providers/common/provision.go

```diff
@@ -1,114 +0,0 @@
-/*
-Copyright 2021 The Kubernetes Authors.
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-    http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
-package common
-
-import (
-	"bufio"
-	"context"
-	"fmt"
-	"io"
-	"regexp"
-	"time"
-
-	"sigs.k8s.io/kind/pkg/exec"
-)
-
-type runContainerOpts struct {
-	waitUntilLogRegexpMatches *regexp.Regexp
-}
-
-type RunContainerOpt func(*runContainerOpts) error
-
-// WithWaitUntilLogRegexpMatches waits until the container log matches the regexp.
-func WithWaitUntilLogRegexpMatches(re *regexp.Regexp) RunContainerOpt {
-	return func(o *runContainerOpts) error {
-		o.waitUntilLogRegexpMatches = re
-		return nil
-	}
-}
-
-// WithWaitUntilSystemdReachesMultiUserSystem waits until the systemd in the container
-// reaches "Multi-User System" target if is using cgroups v2, so that `docker exec` can be
-// executed safely without breaking cgroup v2 hierarchy.
-//
-// This is implemented by grepping `docker logs` with "Reached target .*Multi-User System.*"
-// message from systemd.
-// (we can't use `docker exec` to check whether we are allowed to run `docker exec`, obviously.)
-//
-// Needed for avoiding "ERROR: this script needs /sys/fs/cgroup/cgroup.procs to be empty (for writing the top-level cgroup.subtree_control)"
-// See https://github.com/kubernetes-sigs/kind/issues/2409
```

Comment:
we should keep the comment

Reply:
I renamed this to be even clearer about the purpose and commented this in detail in the latest commit.